File: Go/Algorithms/2315.go
package main

func countAsterisks(s string) int {
	count, flag := 0, false

	for _, ch := range s {
		if ch == '|' {
			flag = !flag
			continue
		}
		if ch == '*' && !flag {
			count++
		}
	}
	return count
}

File: Rust/Algorithms/2315.rs
impl Solution {
    pub fn count_asterisks(s: String) -> i32 {
        let (mut count, mut flag) = (0, false);
        
        for ch in s.chars() {
            if ch == '|' {
                flag = !flag;
                continue;
            }
            if ch == '*' && !flag {
                count += 1;
            }
        }
        count
    }
}